internal/rest/resources/daemon.go

```diff
@@ -0,0 +1,81 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/response"
+
+	"github.com/canonical/microcluster/client"
+	internalClient "github.com/canonical/microcluster/internal/rest/client"
+	internalTypes "github.com/canonical/microcluster/internal/rest/types"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/rest/access"
+	"github.com/canonical/microcluster/rest/types"
+	"github.com/canonical/microcluster/state"
+)
+
+var daemonCmd = rest.Endpoint{
+	Path: "daemon/servers",
+
+	Get: rest.EndpointAction{Handler: daemonServersGet, AccessHandler: access.AllowAuthenticated},
+	Put: rest.EndpointAction{Handler: daemonServersPut, AccessHandler: access.AllowAuthenticated},
+}
+
+func daemonServersGet(s *state.State, r *http.Request) response.Response {
+	return response.SyncResponse(true, s.LocalConfig().GetServers())
+}
+
+func daemonServersPut(s *state.State, r *http.Request) response.Response {
+	req := make(map[string]types.ServerConfig)
+
+	// Parse the request.
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	daemonConfig := s.LocalConfig()
+	daemonConfig.SetServers(req)
+
+	// Persist the configuration changes to file.
+	err = daemonConfig.Write()
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Update the additional listeners.
+	err = s.UpdateServers()
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	cluster, err := s.Cluster(false)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Run the OnDaemonConfigUpdate hook on all other members.
+	remotes := s.Remotes()
+	err = cluster.Query(s.Context, true, func(ctx context.Context, c *client.Client) error {
```

Comment:
My understanding might be off here so please bear with me. AFAIK, `(microcluster.Microcluster).UpdateServers` is the only way we can update server configs during/after cluster initialisation. Once we updated the server configs then the `OnDaemonConfigUpdateHook` would trigger for every member daemon in the cluster. This is not an issue, I'm just trying to validate my understanding since this hook would be a convenient way for me to cache specific server addresses for all cluster members (a requirement in site manager)

Reply:
Yes that is right!